packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts

```diff
@@ -67,7 +73,17 @@ export const processProtectedData = async ({
         OnStatusUpdateFn<ProcessProtectedDataStatuses>
       >(onStatusUpdate);
 
-    const requester = await iexec.wallet.getAddress();
+    let requester = await iexec.wallet.getAddress();
+    if (vUserWhitelist) {
+      const isValidWhitelist = await isDeployedWhitelist(iexec, vUserWhitelist);
+      if (!isValidWhitelist) {
+        throw new Error(
+          `The Whitelist is not valid whitelist contract. It seems the whitelist doesn't respect the IERC734 interface`
+        );
+      }
+      requester = vUserWhitelist;
```

Comment:
call keyHasPurpose on the ERC734 contract to ensure the user is in the whitelist, if not throw an error

Reply:
Fix with the corresponding tests